lib/core-classes/DecodersService.ts

```diff
@@ -1,31 +1,70 @@
-import { PluginContext, EmbeddedSDK, Plugin } from "kuzzle";
+import { DecoderQueue } from './../types/decoders/decodersQueue';
+import { PluginContext, EmbeddedSDK, Plugin, Inflector, PluginImplementationError, PluginApiDefinition } from "kuzzle";
 
 import { Decoder } from "./Decoder";
+import { PayloadHandler, PayloadService } from './PayloadService';
 import { DeviceManagerConfig } from "../DeviceManagerPlugin";
-import { DecoderContent } from '../types/decoders/DecodersContent';
+import { DecoderContent } from "../types/decoders/DecodersContent";
 
 export class DecodersService {
   private config: DeviceManagerConfig;
   private context: PluginContext;
+  private api: PluginApiDefinition;
+  private payloadService: PayloadService;
 
   private decoders: Map<string, Decoder>;
 
   get sdk(): EmbeddedSDK {
     return this.context.accessors.sdk;
   }
 
-  constructor(plugin: Plugin, decoders: Map<string, Decoder>) {
+  constructor(plugin: Plugin, decoders: Map<string, Decoder>, decodersQueue: DecoderQueue[], payloadService: PayloadService) {
     this.config = plugin.config as any;
     this.context = plugin.context;
+    this.api = plugin.api;
+    this.payloadService = payloadService;
 
     this.decoders = decoders;
+
+    decodersQueue.forEach(({ decoder, handler }) => {
+      this.register(decoder, { handler });
+      this.context.log.info(`Register API action "device-manager/payload:${decoder.action}" with decoder "${decoder.constructor.name}" for device "${decoder.deviceModel}"`);
```

Comment:
This log line should be done when the decoder is registered on the plugin by the user otherwise it's more difficult to debug the stacktrace when a faulty decoder has been registered